Chore: fixed lsp config

mmj2023 opened the pull request:
The lsp config has been changed to use new api to get the table for all mason lsp servers.

## Description

<!-- Describe the big picture of your changes to communicate to the maintainers
  why we should accept this pull request. -->

## Related Issue(s)

<!--
  If this PR fixes any issues, please link to the issue here.
  - Fixes #<issue_number>
-->

## Screenshots

<!-- Add screenshots of the changes if applicable. -->

## Checklist

- [ ] I've read the [CONTRIBUTING](https://github.com/LazyVim/LazyVim/blob/main/CONTRIBUTING.md) guidelines.

dpetka2001 commented:
This will not work. Try #6053 instead. Looking for feedback. Or pin the plugin's versions to v1.

mmj2023 commented:
Why could elaborate please?

dpetka2001 commented:
This PR only makes the error go away. It doesn't take into consideration the LSP servers configuration.

fivemtebex commented:
Impressive project structure! The architecture really demonstrates solid design principles.

mmj2023 commented:
@fivemhostingserver I guess you could fix the integration with Lazyvim and tests, if there are any issues, I use a custom config that's heavily inspired by Lazyvim, but I am not a Lazyvim user. So, pardon me if any other area of the code breaks. They have a lot of abstractions that I don't think I need, I think, @dpetka2001 has a better pr than me, see #6053 . I just updated my plug-ins and lspconfig broke, so I fixed it, and when I looked into Lazyvim's repo, it was broken here as well, so I fixed it and made a pr. This is my 1st pr so, I was actually hoping someone would either make a better pr than me or tell me what to do in order to fix mine. And I hadn't looked at the issues yet, so I didn't notice that the issue had already been created and fixes were already given.
